pkg/providers/vsphere/defaults.go

```diff
@@ -77,7 +77,11 @@ func (d *Defaulter) setDefaultTemplateIfMissing(ctx context.Context, spec *Spec,
 
 func (d *Defaulter) setupDefaultTemplate(ctx context.Context, spec *Spec, machineConfig *anywherev1.VSphereMachineConfig) error {
```

Comment:
this might need to change
we might need a different template depending if the machineConfig is used for CP or for a worker node group